cmd/earthly/satellite_cmds.go

```diff
@@ -302,7 +304,15 @@ func (app *earthlyApp) actionSatelliteSelect(cliCtx *cli.Context) error {
 	app.commandName = "satelliteSelect"
 
 	if cliCtx.NArg() != 1 {
-		return errors.New("satellite name is required")
+		if app.cfg.Satellite.Name == "" {
+			app.console.Printf("No satellite selected\n\n")
+			app.console.Printf("To select a satellite:\n\t%s\n", selectUsageText)
```

Comment:
I like the usage text being printed when the command isn't used properly, however, I wonder if the command should return  non-zero exit code.

I'm picturing someone's CI script where it might be useful to know that this failed here and didn't actually select a satellite before it continued on to other things.

Reply:
That's a very good point.. A common case that immediately comes to mind is using an ARG / ENV variable for the select statement and forgetting to set it.

I'll take another look at this later and see about returning a non-zero exit code.

Reply:
Normally it's the fact that an error was returned instead of nil.

I do really like the idea in general of commands printing the usage text when they aren't used properly. Wonder if there's a nice solution that can be reused in all commands. What do you think @alexcb ?

Reply:
It might be possible to tie in to the `cli` package to access the print usage function (whatever it's called), but I don't know offhand (happy to help if needed).

Reply:
Not sure if there is a nicer solution to add that as default behaviour.. We could look.
Worst case scenario, we could make a little closure to do just that.
Roughly, something like this:
```
func wrapAction(usageText string, action func(cliCtx *cli.Context) error) func(cliCtx *cli.Context) error {
	return func(cliCtx *cli.Context) error {
		err := action(cliCtx)
		if err != nil {
			fmt.Printf(usageText)
		}
		return err
	}
}
```

and then we'd use it like this: 
```
{
			Name:        "select",
			Aliases:     []string{"s"},
			Usage:       "Choose which satellite to use to build your app",
			Description: "Choose which satellite to use to build your app",
			UsageText:   selectUsageText,
			Action:      wrapAction(selectUsageText, app.actionSatelliteSelect),
},
```

Reply:
Now returning an error 1 line below.

Reply:
I also just noticed the `cli` package has this function, which might do a very similar thing:
```
// ShowCommandHelpAndExit - exits with code after showing help
func ShowCommandHelpAndExit(c *Context, command string, code int) {
```

Reply:
That could be a nice solution actually, I feel like the code could just look like this instead:

```golang
	if cliCtx.NArg() != 1 {
		cli.ShowCommandHelpAndExit(cliCtx, app.commandName, 1)
	}
```

Not just here, but in a lot of other commands in earthly where we do similar checks. Probably outside the scope of this PR to fix the other ones though.

Reply:
Nice find! I really like that solution.

Reply:
Had to tweak it slightly, but it works nicely!